lib/ansible/module_utils/network/nxos/config/l3_interfaces/l3_interfaces.py

```diff
@@ -92,6 +97,8 @@ def set_config(self, existing_l3_interfaces_facts):
         if config:
             for w in config:
                 w.update({'name': normalize_interface(w['name'])})
+                if w['name'] == 'mgmt0':
+                    self._module.fail_json(msg="The 'mgmt0' interface is not allowed to be managed by this module")
```

Comment:
Instead of failing ignoring mgmt0 with a warning would be a better solution imo. Otherwise every time user executes overridden the module is going to fail?